tracer/src/Datadog.Trace/Configuration/GitMetadataTagsProvider.cs

```diff
@@ -80,6 +84,23 @@ public bool TryExtractGitMetadata([NotNullWhen(true)] out GitMetadata? gitMetada
         }
     }
 
+    private void PropagateGitMetadataToTheProfiler(GitMetadata gitMetadata)
+    {
+        try
+        {
+            // Avoid P/Invoke if the profiler is not ready (for obvious reason)
+            // but also if both repository url and commit sha are empty
+            if (Profiler.Instance.Status.IsProfilerReady && !gitMetadata.IsEmpty)
```

Comment:
Just to double check, is `Profiler.Instance.Status.IsProfilerReady` guaranteed not to throw in all scenarios (including when we're using manual instrumentation (i.e. no native code is loaded)?

Reply:
It's guaranteed not to throw in all scenario.

Reply:
👍